frontend/javascripts/oxalis/view/node_context_menu.js

```diff
@@ -0,0 +1,314 @@
+// @flow
+import * as React from "react";
+import { Menu, notification, Icon, Divider, Popover } from "antd";
+import type { Vector3 } from "oxalis/constants";
+import type { OxalisState, SkeletonTracing } from "oxalis/store";
+import type { Dispatch } from "redux";
+import { connect } from "react-redux";
+import {
+  createEdgeAction,
+  deleteEdgeAction,
+  mergeTreesAction,
+  deleteNodeAction,
+  setActiveNodeAction,
+  createTreeAction,
+  setTreeVisibilityAction,
+} from "oxalis/model/actions/skeletontracing_actions";
+import { setWaypoint } from "oxalis/controller/combinations/skeletontracing_plane_controller";
+import api from "oxalis/api/internal_api";
+import messages from "messages";
+import { getNodeAndTree, findTreeByNodeId } from "oxalis/model/accessors/skeletontracing_accessor";
+import { formatNumberToLength } from "libs/format_utils";
+import { distanceBetweenVectors } from "libs/utils";
+
+type OwnProps = {|
+  nodeContextMenuPosition: [number, number],
+  nodeContextMenuNodeId: ?number,
+  globalPosition: Vector3,
+  rotation: Vector3,
+  hideNodeContextMenu: () => void,
+|};
+
+type DispatchProps = {|
+  deleteEdge: (number, number) => void,
+  createEdge: (number, number) => void,
+  mergeTrees: (number, number) => void,
+  deleteNode: (number, number) => void,
+  setActiveNode: number => void,
+  hideTree: number => void,
+  createTree: () => void,
+|};
+
+type StateProps = {| skeletonTracing: ?SkeletonTracing, datasetScale: Vector3 |};
+
+type Props = {| ...OwnProps, ...StateProps, ...DispatchProps |};
+type NodeContextMenuOptionsProps = {| ...Props, nodeContextMenuNodeId: number |};
+type NoneNodeContextMenuProps = {| ...Props |};
+
+const NAVBAR_HEIGHT = 48;
```

Comment:
Can we get rid of this constant somehow? Maybe compute it ad-hoc via the DOM element? Seems very brittle in case the navbar height changes at some point.

Reply:
I think, you can completely get rid of this if you move the contextmenu within `tracing_layout_view.js` into the `<div id={canvasAndLayoutContainerID} style={{ position: "relative" }}>` component. That way, the absolute position of the context menu is relative to that child (which is positioned below the navbar).

Reply:
The problem here is that the position from the event is relative to the whole page but the context menu is positioned within the main part of the tracing view. This container is relative and has the y offset that is equal to the navbar height. That's why the context menu is positioned too low if the navbar height is not subtracted.

Reply:
But placing it at the same hierarchy level as the nml upload container works fine as the container has the position set to relative :+1: